src/sdk/namespace_fs.js

```diff
@@ -79,6 +79,45 @@ function sort_entries_by_name(a, b) {
     return 0;
 }
 
+function _get_mtime_from_filename(filename) {
+    if (!filename.includes('_mtime-')) {
+        // Latest file wont have time suffix which will push the latest
+        // object last in the list. So to keep the order maintained,
+        // returning the latest time. Multiplying with 1e6 to provide
+        // nano second precision
+        return BigInt(Date.now() * 1e6);
+    }
+    const file_parts = filename.split('-');
+    return size_utils.string_to_bigint(file_parts[file_parts.length - 3], 36);
+}
+
+function _get_filename(file_name) {
+    if (file_name.includes('_mtime-')) {
+        return file_name.substring(0, file_name.indexOf('_mtime-'));
+    }
+    return file_name;
+}
+/**
+ * @param {fs.Dirent} a
+ * @param {fs.Dirent} b
+ * @returns {Number}
+ */
+function sort_entries_by_name_and_time(a, b) {
+    const a_name = _get_filename(a.name);
+    const b_name = _get_filename(b.name);
+    if (a_name === b_name) {
+        const a_mtime = _get_mtime_from_filename(a.name);
+        const b_mtime = _get_mtime_from_filename(b.name);
+        if (b_mtime < a_mtime) return -1;
```

Comment:
when a_mtime < b_mtime shouldn't we return -1?

Reply:
With this, the list will be arranged in the latest first order

Reply:
right, can you add a comment on it please?

Reply:
sure